pca-server/src/pca/pca-aws-sf-process-turn-by-turn.py

```diff
@@ -1297,123 +1276,94 @@ def create_playback_mp3_audio(self, audio_uri):
+        json_filepath = TMP_DIR + '/' + sf_event["transcriptUri"].split("/")[-1]
+        if sf_event["transcriptUri"].startswith("https"):
+            # HTTPS URI came from Transcribe, so https://<region>/<bucket>/<key>
+            transcriptResultsKey = "/".join(sf_event["transcriptUri"].split("/")[4:])
+        else:
+            # S3 URI came from Transcribe, so s3://<bucket>/<key>
+            transcriptResultsKey = "/".join(sf_event["transcriptUri"].split("/")[3:])
 
         # Now download - this has been known to get a "404 HeadObject Not Found",
         # which makes no sense, so if that happens then re-try in a sec.  Only once.
         s3Client = boto3.client('s3')
         try:
-            s3Client.download_file(outputS3Bucket, transcriptResultsKey, json_filepath)
+            s3Client.download_file(output_bucket, transcriptResultsKey, json_filepath)
         except:
             time.sleep(3)
-            s3Client.download_file(outputS3Bucket, transcriptResultsKey, json_filepath)
+            s3Client.download_file(output_bucket, transcriptResultsKey, json_filepath)
 
-        # Before we process, let's load up any required simply entity map
+        # Load in the JSON file for processing, and set our language codes for the file and Comprehend
+        self.asr_output = json.load(open(Path(json_filepath).absolute(), "r", encoding="utf-8"))
+
+        # Before we process, let's load up any required simply entity map, which needs the base language code
+        self.set_comprehend_language_code()
         self.load_simple_entity_string_map()
 
         # Now create turn-by-turn diarisation, with associated sentiments and entities
-        self.speechSegmentList = self.create_turn_by_turn_segments(json_filepath)
+        self.speechSegmentList = self.create_turn_by_turn_segments(sf_event)
 
         # Update our results data structures, generate JSON results and save them to S3
         self.push_turn_by_turn_results()
 
-        # Write out the JSON data to our S3 location
-        interim_results_file_key = TEMP_OUTPUT_KEY + '/' + self.jsonOutputFilename
-        json_output = self.pca_results.write_results_to_s3(outputS3Bucket, interim_results_file_key)
+        # Write out the JSON data back to our interim S3 location
+        json_output = self.pca_results.write_results_to_s3(bucket=output_bucket, object_key=sf_event["interimResultsFile"])
 
-        # Index transcript in Kendra, if transcript search is enabled
+        # Index transcript in Kendra, if transcript search is enable
         kendraIndexId = cf.appConfig[cf.CONF_KENDRA_INDEX_ID]
         if kendraIndexId != "None":
-            analysisUri = f"{cf.appConfig[cf.CONF_WEB_URI]}dashboard/parsedFiles/{self.jsonOutputFilename}"
+            analysisUri = f"{cf.appConfig[cf.CONF_WEB_URI]}dashboard/parsedFiles/{json_filepath.split('/')[-1]}"
             transcript_with_markers = prepare_transcript(json_filepath)
```

Comment:
prepare_transcript() function fails with streaming post-call output files...   Because it currently depends on the word level 'Items' which are missing in the streaming post-call output files.. 
I'll have a go at refactoring tomorrow.
```
{
  "errorMessage": "'Items'",
  "errorType": "KeyError",
  "stackTrace": [
    "  File \"/var/task/pca-aws-sf-process-turn-by-turn.py\", line 1377, in lambda_handler\n    transcribeParser.parse_transcribe_file(sf_data)\n",
    "  File \"/var/task/pca-aws-sf-process-turn-by-turn.py\", line 1354, in parse_transcribe_file\n    transcript_with_markers = prepare_transcript(json_filepath)\n",
    "  File \"/var/task/pcakendrasearch.py\", line 77, in prepare_transcript\n    return prepare_transcript_callanalytics(transcript)\n",
    "  File \"/var/task/pcakendrasearch.py\", line 45, in prepare_transcript_callanalytics\n    items = turn[\"Items\"]\n"
  ]
}
```
Affected code
```
def prepare_transcript_callanalytics(transcript):
    print(f"prepare_transcript_callanalytics(...)")
    turns = transcript["Transcript"]
    txt = ""
    sentence = ""
    for turn in turns:
        items = turn["Items"]
```